src/jetstream/datastore/database_cf_config.go

```diff
@@ -163,9 +181,15 @@ func findDatabaseConfigurationFromUri(uri string) (string, string, string, int,
 	username := md["username"]
 	password := md["password"]
 	host := md["host"]
-	port, _ := strconv.Atoi(fmt.Sprintf("%v", md["port"]))
+	portStr := fmt.Sprintf("%v", md["port"])
+	var port int
+	if portStr != "<nil>" {
+		port, _ = strconv.Atoi(portStr)
+	} else {
+		port = 0
```

Comment:
Is this actually an error case? If we return 0 as the port, that's not going to work?

Perhaps we return 0 and in the code above, if the port is 0 we use the default port for the database provider - 3306 for MySQL and 5432 for postgres.
Should we be returning